tests/Graph/MockGraphProvider.h

```diff
@@ -67,9 +67,11 @@ class MockGraphProvider {
    public:
     class Vertex {
      public:
-      explicit Vertex(VertexType v) : _vertex(v){};
+      explicit Vertex(VertexType v) : _vertex(v) { _depth = 0; };
```

Comment:
Also here I think the Vertex does not have to know it's depth, the Step is enough

Reply:
Changed and moved to parent class:  BaseStep.